administrator/components/com_joomgallery/helpers/file.php

```diff
@@ -367,301 +383,409 @@ public static function resizeImage(&$debugoutput, $src_file, $dest_file, $usefor
+          foreach ($src_frames as $key => $frame)
+          {
+            $src_frames[$key]['image'] = imagerotate($src_frames[$key]['image'], $angle, 0);
+          }
         }
-
-        $dst_img = imagecreatetruecolor($destWidth, $destHeight);
-
-        if($config->jg_fastgd2thumbcreation == 0)
+        foreach ($src_frames as $key => $frame)
         {
-          if(!is_null($offsetx) && !is_null($offsety))
+          if($config->jg_fastgd2thumbcreation == 0)
+          // use normal GD2 for resizing
           {
-            imagecopyresampled( $dst_img, $src_img, 0, 0, $offsetx, $offsety,
-                                $destWidth, (int)$destHeight, $srcWidth, $srcHeight);
+            if(!is_null($offsetx) && !is_null($offsety))
+            {
+              imagecopyresampled( $dst_frames[$key], $src_frames[$key]['image'], 0, 0, $offsetx, $offsety,
+                                  $destWidth, (int)$destHeight, $srcWidth, $srcHeight);
+            }
+            else
+            {
+              imagecopyresampled( $dst_frames[$key], $src_frames[$key]['image'], 0, 0, 0, 0, $destWidth,
+                                  (int)$destHeight, $srcWidth, $srcHeight);
+            }
           }
           else
+          // use fast GD2 for resizing
           {
-            imagecopyresampled( $dst_img, $src_img, 0, 0, 0, 0, $destWidth,
-                                (int)$destHeight, $srcWidth, $srcHeight);
+            if(!is_null($offsetx) && !is_null($offsety))
+            {
+              $dst_frames[$key] = JoomFile::fastImageCopyResampled( $dst_frames[$key], $src_frames[$key]['image'], 0, 0, $offsetx, $offsety,
+                                                                    $destWidth, (int)$destHeight, $srcWidth, $srcHeight, 3,$imginfo);
+            }
+            else
+            {
+              $dst_frames[$key] = JoomFile::fastImageCopyResampled( $dst_frames[$key], $src_frames[$key]['image'], 0, 0, 0, 0, $destWidth,
+                                                                    (int)$destHeight, $srcWidth, $srcHeight, 3,$imginfo);
+            }
           }
         }
+        // write resized image to file
+        if ($anim && $imginfo['animation'] && $imginfo['type'] == 'GIF')
+        {
+          JLoader::register('GifCreator', JPATH_COMPONENT_ADMINISTRATOR . '/helpers/GifCreator.php');
+          $gc = new GifCreator();
+          $gc->create($dst_frames, $gfe->getFrameDurations(), 0);
+          $success = file_put_contents($dest_file, $gc->getGif());
+        }
         else
         {
-          if(!is_null($offsetx) && !is_null($offsety))
-          {
-            JoomFile::fastImageCopyResampled( $dst_img, $src_img, 0, 0, $offsetx, $offsety,
-                                              $destWidth, (int)$destHeight, $srcWidth, $srcHeight);
-          }
-          else
+          $success = JoomFile::imageWriteFrom_GD($dest_file,$dst_frames,$dest_qual,$dest_imgtype);
```

Comment:
If it is no animation, write resized image with the external function JoomFile::imageWriteFrom_GD. It writes the resized frames to one destination file of the desired quality and file-type.